src/server/scripts/EasternKingdoms/ZulGurub/boss_thekal.cpp

```diff
@@ -15,443 +15,490 @@
+    {
+        Talk(EMOTE_ZEALOT_DIES);
 
-                context.Repeat(15s, 20s);
-            }).Schedule(6s, [this](TaskContext context) {
-                DoCastVictim(SPELL_DISARM);
-                context.Repeat(15s, 25s);
-            });
+        if (Creature* thekal = instance->GetCreature(DATA_THEKAL))
+        {
+            thekal->AI()->SetData(ACTION_RESSURRECT, DATA_LORKHAN);
         }
+    }
 
-        void JustDied(Unit* /*killer*/) override
+    void UpdateAI(uint32 diff) override
+    {
+        if (me->IsInCombat() && !UpdateVictim())
         {
-            Talk(EMOTE_ZEALOT_DIES);
-
-            if (Creature* thekal = instance->GetCreature(DATA_THEKAL))
-            {
-                thekal->AI()->SetData(ACTION_RESSURRECT, DATA_LORKHAN);
-            }
+            return;
         }
-
-        void UpdateAI(uint32 diff) override
+        else if (me->IsInCombat())
         {
-            if (me->GetReactState() != REACT_PASSIVE && !UpdateVictim())
-                return;
-
             _scheduler.Update(diff,
-                std::bind(&ScriptedAI::DoMeleeAttackIfReady, this));
+            std::bind(&BossAI::DoMeleeAttackIfReady, this));
+        }
+        else
+        {
+            _scheduler.Update(diff);
         }
+    }
 
-        private:
-            TaskScheduler _scheduler;
-    };
+    private:
+        TaskScheduler _scheduler;
+};
 
-    CreatureAI* GetAI(Creature* creature) const override
+struct npc_zealot_zath : public ScriptedAI
+{
+    npc_zealot_zath(Creature* creature) : ScriptedAI(creature)
     {
-        return GetZulGurubAI<npc_zealot_lorkhanAI>(creature);
+        instance = creature->GetInstanceScript();
     }
-};
 
-class npc_zealot_zath : public CreatureScript
-{
-public:
-    npc_zealot_zath() : CreatureScript("npc_zealot_zath") { }
+    InstanceScript* instance;
 
-    struct npc_zealot_zathAI : public ScriptedAI
+    void Reset() override
     {
-        npc_zealot_zathAI(Creature* creature) : ScriptedAI(creature)
-        {
-            instance = creature->GetInstanceScript();
-        }
-
-        InstanceScript* instance;
+        _scheduler.CancelAll();
 
-        void Reset() override
+        _scheduler.SetValidator([this]
         {
-            _scheduler.CancelAll();
+            return !me->HasUnitState(UNIT_STATE_CASTING);
+        });
+
+        // emote idle loop
+        _scheduler.Schedule(5s, 25s, [this](TaskContext context) {
+            // pick a random emote from the list of available emotes
+            me->HandleEmoteCommand(
+                RAND(
+                    EMOTE_ONESHOT_TALK,
+                    EMOTE_ONESHOT_BEG,
+                    EMOTE_ONESHOT_YES
+                )
+            );
+            context.Repeat(5s, 25s);
+        });
+    }
 
-            _scheduler.SetValidator([this]
+    void JustEngagedWith(Unit* /*who*/) override
+    {
+        _scheduler.CancelAll();
+
+        _scheduler.Schedule(13s, [this](TaskContext context) {
```

Comment:
bracket for scheduler needs to be on new line to follow convention. sorry

Reply:
`azerothcore-python` wen?